pkg/workloadmeta/collectors/internal/containerd/image_sbom_trivy.go

```diff
@@ -44,9 +44,7 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 	trivyConfiguration.ContainerdAccessor = func() (cutil.ContainerdItf, error) {
 		return c.containerdClient, nil
 	}
-	trivyConfiguration.CacheProvider = func() (cache.Cache, error) {
-		return trivy.NewLocalCache(config.Datadog.GetString("container_image_collection.sbom.cache_directory"))
-	}
+	trivyConfiguration.CacheProvider = cacheProvider(true)
```

Comment:
@lebauce If we want to use Badger for the security-agent too we'll need to move this call to the `DefaultCollectorConfig` (called some lines above).